best-practices-zarr-ndarray.md

```diff
@@ -0,0 +1,741 @@
+          "cf:standard_name": "cloud_area_fraction",
+          "unit": "m2 m-2",
+          "cube:dimension_refs": ["time", "cell"],
+          "cube:type": "data",
+        },
+        {
+          "name": "hfls",
+          "description": "surface upward latent heat flux",
+          "cf:standard_name": "surface_upward_latent_heat_flux",
+          "unit": "W m-2",
+          "cube:dimension_refs": ["time", "cell"],
+          "cube:type": "data",
+        },
+        {
+          "name": "pr",
+          "description": "precipitation flux",
+          "cf:standard_name": "precipitation_flux",
+          "unit": "kg m-2 s-1",
+          "cube:dimension_refs": ["time", "cell"],
+          "cube:type": "data",
+        },
+        {
+          "name": "tas",
+          "description": "near-surface air temperature",
+          "cf:standard_name": "air_temperature",
+          "unit": "K",
+          "cube:dimension_refs": ["time", "cell"],
+          "cube:type": "data",
+        },
+        {
+          "name": "ta",
+          "description": "air temperature",
+          "cf:standard_name": "air_temperature",
+          "unit": "K",
+          "cube:dimension_refs": ["time", "pressure", "cell"],
+          "type": "data",
+        }
+      ]
+    }
+  }
+}
+```
+
+
+**Key Points**:
+
+- Single asset represents an entire xarray Dataset
+- Multiple dimension types: temporal, spatial (non-standard grid), vertical
+- Variables have different dimensional structures (2D vs 3D)
+- CF standard names are preserved using a `cf:` prefix
+- The `bands` array provides human-readable variable descriptions
+- Non-geographic spatial reference system (HEALPix) is indicated
+
+### Virtual Zarr with Reference Files
+
+**Context**: Kerchunk and similar tools create virtual Zarr stores that reference existing data files (e.g., NetCDF) without copying data. This enables Zarr-like access to legacy formats.
+
+**STAC Representation**:
+
+```json
+{
+  "type": "Feature",
+  "stac_version": "1.1.0",
+  "id": "CMIP6_ScenarioMIP_NCAR_CESM2",
+  "properties": {
+    "start_datetime": "2015-01-01T00:00:00Z",
+    "end_datetime": "2024-12-31T23:59:59Z"
+  },
+  "links": [
+    {
+      "rel": "store",
+      "href": "https://storage.example.com/kerchunk/CMIP6_CESM2_rlus_kr1.0.json",
+      "type": "application/json+zstd",
+      "title": "Kerchunk Reference File"
+    }
+  ],
+  "assets": {
+    "source_data": {
+      "href": "https://storage.example.com/data/CMIP6/rlus_day_CESM2-WACCM_ssp585_gn_20150101-20241231.nc",
+      "type": "application/netcdf",
+      "title": "Original NetCDF file",
+      "roles": ["source"]
+    }
+  }
+}
+```
+
+**Key Points**:
+
+- The kerchunk reference file is considered as the data store and thus is reference as a link with `rel: store`
+- Assets include both the reference file and source data
+- Role `"reference"` indicates virtual/indirect data access
+- Role `"source"` indicates the underlying data files
```

Comment:
Don't think these points match the example above, since the Assets don't include a reference file. Should Virtual Zarr files follow the same principle of the normal Zarr files if they reference a group of arrays they belong in Assets if they are a Zarr store they should be included in the links?